### PR TITLE
Refactor script wrappers to minimize code duplication

### PR DESCRIPTION
**Pull request by ibratoev**
Create a base class for Pipeable wrappers and extract logic there. Simplify the JS code wrappers and extract reusable function.

Create real Page and Element wrappers and add the same test as Pipeable base classes.

### CODE REVIEW

File: Sources/PipeableSDK/Script/Fakes.swift [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        if elementId.isEmpty {
+            throw PipeableError.elementNotFound
+        }
+        print("Clicking on element with id: \(elementId) ...")
+        try await Task.sleep(nanoseconds: 1_000_000_000)
+        print("Click completed!")
+    }
+}
+
+@objc protocol FakeElementJSExport: JSExport {
```
Comment (pdobrev) [inline Sources/PipeableSDK/Script/Fakes.swift:46]: Isn't this the same protocol as `ElementJSExport`? Do we need a mock protocol as well?

File: Sources/PipeableSDK/Script/Fakes.swift [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    func clickWithCompletion(_ completionHandler: JSValue) {
+        doWithCompletion(completionHandler) {
+            try await self.element.click()
+            self.successFullClicksClount += 1
+            return CompletionResult.success(nil)
+        }
+    }
+}
+
+@objc protocol FakePageJSExport: JSExport {
```
Comment (pdobrev) [inline Sources/PipeableSDK/Script/Fakes.swift:68]: Same as above

File: Sources/PipeableSDK/ScriptContext.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        super.init()
+    }
+
+    func scriptStarted() {
+        dispatchGroup.enter()
+    }
+
+    func scriptEnded() {
+        Task {
+            self.dispatchGroup.leave()
```
Comment (pdobrev) [inline Sources/PipeableSDK/ScriptContext.swift:22]: Why does this need to be run in a Task detached from the executing thread? Do we need to post these on a queue or this kind of fire & forget works?

**Review by pdobrev — APPROVED** [review]
Added a couple of comments mostly as questions, but im also good to merge as-is and fix these incrementally as we keep adding more tests.

File: Sources/PipeableSDK/ScriptContext.swift (hunk 3 above)
Comment (ibratoev) [reply Sources/PipeableSDK/ScriptContext.swift:22]: Good question... To be honest, there should be no problem running it without a Task...

Comment (ibratoev) [reply Sources/PipeableSDK/ScriptContext.swift:22]: I will try it and remove it if I don't find out why it needs to be like that :)

File: Sources/PipeableSDK/Script/Fakes.swift (hunk 2 above)
Comment (ibratoev) [reply Sources/PipeableSDK/Script/Fakes.swift:68]: Those will be deleted soon. The goal is not to keep them in sync with the real one.

**Pull request merged**